crates/llm-ls/src/retrieval.rs

```diff
@@ -459,31 +480,63 @@ impl SnippetRetriever {
             let window = lines[start_line..end_line].to_vec();
             let snippet = window.join("\n");
             if snippet.is_empty() {
-                continue;
+                    debug!("snippet {file_url}[{start_line}, {end_line}] empty");
+                    continue;
             }
+            snippets.push(Snippet{ file_url: file_url.clone().into(), code: snippet, start_line, end_line });
+        }
+        {
+            let nb_snippets = snippets.len();
+            let steps = self.window_step;
+            debug!("Found {nb_snippets} snippets in {file_url}: {start}, {end}, {steps}");
+        }
 
-            let mut encoding = self.tokenizer.encode(snippet.clone(), true)?;
-            encoding.truncate(512, 1, TruncationDirection::Right);
-            let result = self.generate_embedding(encoding, self.model.clone()).await;
-            let embedding = match result {
-                Ok(e) => e,
+        // Group by length to reduce padding effect
+        snippets.sort_by(|first, second| first.code.len().cmp(&second.code.len()));
+        // TODO improvements to compute an efficient batch size:
```

Comment:
```suggestion
        // TODO: improvements to compute an efficient batch size:
```
My editor won't pickup on the `TODO` w/out the semicolon 😄